runtime/src/main/java/org/corfudb/security/tls/TlsUtils.java

```diff
@@ -4,134 +4,102 @@
-                                       String keyStore,
-                                       Consumer<Exception> keyStoreException,
+    public static SslContext enableTls(boolean isServer,
+                                       String keyStorePath,
                                        String ksPasswordFile,
-                                       Consumer<Exception> ksPasswordFileException,
-                                       String trustStore,
-                                       Consumer<Exception> trustStoreException,
-                                       String tsPasswordFile,
-                                       Consumer<Exception> tsPasswordFileException) {
+                                       String trustStorePath,
+                                       String tsPasswordFile) throws SSLException {
         // Get the key store password
         String ksp = "";
         if (ksPasswordFile != null) {
             try {
-                ksp = (new String(Files.readAllBytes(Paths.get(ksPasswordFile))))
-                        .trim();
-            } catch (Exception e) {
-                keyStoreException.accept(e);
-                return null;
+                ksp = (new String(Files.readAllBytes(Paths.get(ksPasswordFile)))).trim();
+            } catch (IOException e) {
+                String errorMessage = "Unable to read key store password file " + ksPasswordFile + ".";
+                log.error(errorMessage, e);
+                throw new SSLException(errorMessage, e);
             }
         }
+
         // Get the key store
-        KeyStore ks = null;
-        if (keyStore != null) {
-            try (FileInputStream fis = new FileInputStream(keyStore)) {
-                ks = KeyStore.getInstance(KeyStore.getDefaultType());
-                ks.load(fis, ksp.toCharArray());
-            } catch (Exception e) {
-                ksPasswordFileException.accept(e);
-                return null;
-            }
+        if (keyStorePath == null) {
+            String errorMessage = "Certificate key store is not specified.";
+            log.error(errorMessage);
+            throw new SSLException(errorMessage);
         }
 
-        // Get the trust store password
-        String tsp = "";
-        if (tsPasswordFile != null) {
-            try {
-                tsp = (new String(Files.readAllBytes(Paths.get(tsPasswordFile))))
-                        .trim();
-            } catch (Exception e) {
-                trustStoreException.accept(e);
-                return null;
-            }
+        KeyManagerFactory kmf;
+        try {
+            FileInputStream fis = new FileInputStream(keyStorePath);
+            KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+            ks.load(fis, ksp.toCharArray());
+
+            kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(ks, ksp.toCharArray());
+        } catch (IOException e) {
```

Comment:
there is a lot of boilerplate code here. Why not make a small private method that takes error string as parameter ?

Reply:
Fixed. Did some refactor, and added some unit tests for them.